src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -819,5 +832,37 @@ private static ILogger GetLogger(HttpContext httpContext)
                 return loggerFactory.CreateLogger(typeof(RequestDelegateFactory));
             }
         }
+
+        private static void EnsureRequestTaskOfNotNull<T>(Task<T> task)
+        {
+            if (task is null)
+            {
+                throw new InvalidOperationException("Task Result should not be null");
+            }
+        }
+
+        private static void EnsureRequestTaskNotNull(Task task)
+        {
+            if (task is null)
+            {
+                throw new InvalidOperationException("Task Result should not be null");
+            }
+        }
+
+        private static void EnsureRequestValueTaskOfNotNull<T>(ValueTask<T> task)
+        {
+            if (task.Result is null)
```

Comment:
This isn't the right check. .Result will cause the task to block.

Reply:
It would be **`task.Equals(default(ValueTask<T>))`** ?. In this case, are we sure that we are also testing that T is not null ?

Reply:
We don't need to handle the T being null. We'll write out a null JSON response. WE need to handle null IResult because we de-reference it.

Reply:
Ok, I understand 🙂